herddb-collections/src/main/java/herddb/collections/TmpMapImpl.java

```diff
@@ -163,39 +187,106 @@ public long estimateCurrentMemoryUsage() {
     }
 
     @Override
-    public V get(K key) throws Exception {
+    public V get(K key) throws CollectionsException {
         byte[] serializedKey = keySerializer.apply(key);
         return (V) executeGet(serializedKey, tmpTableName);
     }
 
     @Override
-    public boolean containsKey(K key) throws Exception {
+    public boolean containsKey(K key) throws CollectionsException {
         byte[] serializedKey = keySerializer.apply(key);
         return executeContainsKey(serializedKey, tmpTableName);
     }
 
-    private Object executeGet(byte[] serializedKey, String tmpTableName) throws StatementExecutionException, Exception {
-        GetStatement get = new GetStatement(TableSpace.DEFAULT, tmpTableName, Bytes.from_array(serializedKey), null,
-                false);
-        GetResult getResult = (GetResult) tableSpaceManager.executeStatement(get,
-                new StatementEvaluationContext(),
-                herddb.model.TransactionContext.NO_TRANSACTION);
-        if (!getResult.found()) {
-            return null;
-        } else {
-            return valuesSerializer
-                    .deserialize(getResult.getRecord().value);
+    @Override
+    public void forEach(BiSink<K, V> sink) throws CollectionsException, InvocationTargetException {
```

Comment:
Why did you used java.lang.reflect.InvocationTargetException and not a CollectionsException specialization like 
"CollectionsInvocationException" or "SinkInvocationException"?

Reply:
Because it is the same practice we are following un MagNews API.
InvocationTargetException usually means that the code I am calling threw an Exception.
I am not so a bug fan of it. I will change